mobile/src/main/java/org/openhab/habdroid/background/ItemsControlsProviderService.kt

```diff
@@ -0,0 +1,337 @@
+            "lawnmower" -> DeviceTypes.TYPE_MOWER
+            "humidity" -> DeviceTypes.TYPE_HUMIDIFIER
+            "heating", "temperature" -> DeviceTypes.TYPE_HEATER
+            "poweroutlet" -> DeviceTypes.TYPE_OUTLET
+            "door", "frontdoor" -> DeviceTypes.TYPE_DOOR
+            "alarm" -> DeviceTypes.TYPE_SECURITY_SYSTEM
+            "water" -> DeviceTypes.TYPE_SHOWER
+            "garage", "garagedoor", "garage_detached", "garage_detached_selected" -> DeviceTypes.TYPE_GARAGE
+            else -> when {
+                item.isOfTypeOrGroupType(Item.Type.Rollershutter) || item.tags.contains(Item.Tag.Blind) ->
+                    DeviceTypes.TYPE_BLINDS
+                item.tags.contains(Item.Tag.Thermostat) || item.tags.contains(Item.Tag.TargetTemperature) ->
+                    DeviceTypes.TYPE_HEATER
+                item.tags.contains(Item.Tag.Lighting) -> DeviceTypes.TYPE_LIGHT
+                else -> DeviceTypes.TYPE_UNKNOWN
+            }
+        }
+
+    private fun createRangeTemplate(item: Item, format: String): RangeTemplate {
+        val currentValue = item.state?.asNumber?.value ?: 0F
+        val minimum = min(currentValue, item.minimum?.toFloat() ?: 0F)
+        val maximum = max(currentValue, item.maximum?.toFloat() ?: 100F)
+        return RangeTemplate(
+            item.name, minimum, maximum,
+            currentValue, item.step?.toFloat() ?: 1F,
+            format
+        )
+    }
+
+    private fun maybeCreateControl(item: Item, stateful: Boolean): Control? {
+        if (item.label.isNullOrEmpty() || item.readOnly) return null
+
+        if (stateful) {
+            val controlTemplate = when (item.type) {
+                Item.Type.Switch -> ToggleTemplate(
+                    item.name,
+                    ControlButton(item.state?.asBoolean ?: false, getString(R.string.nfc_action_toggle))
+                )
+                Item.Type.Dimmer, Item.Type.Color -> ToggleRangeTemplate(
+                    "${item.name}_toggle",
+                    ControlButton(item.state?.asBoolean ?: false, getString(R.string.nfc_action_toggle)),
+                    createRangeTemplate(item, "%.0f%%")
+                )
+                Item.Type.Rollershutter -> createRangeTemplate(item, "%.0f%%")
+                Item.Type.Number -> createRangeTemplate(
+                    item,
+                    item.state?.asNumber?.unit?.let { "%.0f $it" } ?: "%.0f"
+                )
+                else -> return null
+            }
+
+            return Control.StatefulBuilder(item.name, mainActivityPendingIntent)
+                .setTitle(item.label)
+                .setDeviceType(getDeviceType(item))
+                .setControlTemplate(controlTemplate)
+                .setStatus(Control.STATUS_OK)
+                .build()
+        } else {
+            return when (item.type) {
+                Item.Type.Switch, Item.Type.Dimmer, Item.Type.Color, Item.Type.Rollershutter, Item.Type.Number ->
```

Comment:
This list of valid item types has to be maintained here and in the assignment of `controlTemplate` above. I'd assign `controlTemplate` unconditionally and keep the `return null` for unsupported item types there.
Then this list of types can be removed here.